### DISCUSSION TITLE
Paymments, Stripe and description

### DISCUSSION THREAD

**Discussion by StuartMcD** [solspace/craft-freeform#1556] [Questions & Answers] [answered]
So I've wired up a freeform/Stripe integration and have it working 🎉

The issue I'm now wondering is, with all these payments being listed in Stripe, they all share the same description 'Payment from \"Stripe payments\" form' - is there any way I can also attach (or override) bespoke info? IE. description: "Steve Tester, Red Team, deposit"

**Comment by StuartMcD** [comment]
Stat that, found where I can set that.

**Comment by gustavs-gutmanis** [comment, marked as answer]
Hey @StuartMcD,

Yes you should be able to customize the title via events, if you're comfortable with that.

There's an event [`Stripe::EVENT_UPDATE_PAYMENT_METADATA`](https://docs.solspace.com/craft/freeform/v5/developer/events/stripe-payments/#when-updating-payment-metadata) which lets you pass your own metadata to Stripe. You would need to pass your data in the  `description` key, to have it update the entry to your liking.

The [`UpdateMetadataEvent`](https://github.com/solspace/craft-freeform/blob/v5/packages/plugin/src/Integrations/PaymentGateways/Events/UpdateMetadataEvent.php) has references to the `Form` and `Integration` objects, however the `Submission` object isn't present at this time, it becomes available once the stripe payment is processed. For that there's another event, but I'm looking in the code and I see that we're not allowing updating of `description` from there, which is an oversight that we will correct.

**Comment by StuartMcD** [comment]
LOL, I understand the concept of events but woefully shy of actual chops to do it.

I thought I may have been able to use the 'description' field in the form to pass {{ submission.var }} but that hasn't panned out - which must align with something you just wrote :P

I might need to hire you 🥺

**Reply by gustavs-gutmanis** [reply]
If it's not super pressing, we could implement an input in the UI where you would be able to use twig to shape the description to your desires, but we have some more pressing items in the backlog first, so I cannot provide a reliable estimate on when we'd get to it.

**Comment by croxton** [comment]
Hi, was there any progress on allowing the Stripe payment description to contain submission data? In Freeform 4.x it was possible to output the value of a field in the description like this `Reference: {reference}` (where `reference` is the handle of a text field). But this no longer works in v5. `{{submission.reference}}` also doesn't work.

I also tried adding listening to `EVENT_AFTER_UPDATE_PAYMENT_METADATA` in a custom module, but not having any luck:

```php
Event::on(
    Stripe::class,
    Stripe::EVENT_AFTER_UPDATE_PAYMENT_METADATA,
    static function (UpdateMetadataEvent $event) {
        // $event->getSubmission() doesn't exist :(
        $submission = $event->getForm()->getSubmission();
        if ($submission) {
            // Doesn't work
            $event->addData('description', $submission->reference);
        }
    }
);
```

Any ideas how to do this? The accountant for this project needs to be able to see the reference number in the description, otherwise it's hard for them to match a payment in Stripe to a form submission.

**Comment by croxton** [comment]
For anyone finding this, this is how you would add a form value to the description as it appears in the Stripe transaction log:

```php
 Event::on(
    Stripe::class,
    Stripe::EVENT_UPDATE_PAYMENT_METADATA,
    static function (UpdateMetadataEvent $event) {
        $form = $event->getForm();
        $reference = $form->get('reference')->getValue();
        $event->addData('description', $reference);
    }
);
```

**Reply by kjmartens** [reply]
Thanks for sharing the solution, @croxton 🙂
